Composer/packages/adaptive-form/src/components/fields/IntellisenseFields.tsx

```diff
@@ -0,0 +1,173 @@
+          onChange={(newValue) => onValueChanged(newValue || '')}
+          onClick={onClickTextField}
+          onKeyDown={onKeyDownTextField}
+          onKeyUp={onKeyUpTextField}
+        />
+      )}
+    </Intellisense>
+  );
+};
+
+export const IntellisenseNumberField: React.FC<FieldProps<string>> = function IntellisenseNumberField(props) {
+  const { id, value = '', onChange, uiOptions, focused: defaultFocused } = props;
+
+  const completionListOverrideResolver = (value: string) => {
+    if (value === '') {
+      return <ExpressionSwitchWindow type={'Number'} onSwitchToExpression={() => onChange('=')} />;
+    } else {
+      return null;
+    }
+  };
+
+  return (
+    <Intellisense
+      completionListOverrideResolver={completionListOverrideResolver}
+      focused={defaultFocused}
+      id={`intellisense-${id}`}
+      scopes={uiOptions.intellisenseScopes || []}
+      url={getIntellisenseUrl()}
+      value={value}
+      onBlur={props.onBlur}
+      onChange={onChange}
+    >
+      {({ textFieldValue, focused, onValueChanged, onKeyDownTextField, onKeyUpTextField, onClickTextField }) => (
+        <NumberField
+          {...props}
+          focused={focused}
+          id={id}
+          value={textFieldValue}
+          onBlur={undefined} // onBlur managed by Intellisense
+          onChange={(newValue) => onValueChanged(newValue || 0)}
+          onClick={onClickTextField}
+          onKeyDown={onKeyDownTextField}
+          onKeyUp={onKeyUpTextField}
+        />
+      )}
+    </Intellisense>
+  );
+};
+
+export const IntellisenseJSONField: React.FC<FieldProps<string>> = function IntellisenseJSONField(props) {
+  const { id, value = '', onChange, focused: defaultFocused, schema } = props;
+
+  const completionListOverrideResolver = (value: any) => {
+    if (typeof value === 'object' && Object.keys(value).length === 0) {
+      return (
+        <ExpressionSwitchWindow
+          type="Object"
+          onSwitchToExpression={() => {
+            onChange('=');
+          }}
+        />
+      );
+    } else if (Array.isArray(value) && value.length === 0) {
+      return (
+        <ExpressionSwitchWindow
+          type="Array"
+          onSwitchToExpression={() => {
+            onChange('=');
+          }}
+        />
+      );
+    }
+    return null;
+  };
+
+  const defaultValue = schema.type === 'object' ? {} : [];
+
+  return (
+    <Intellisense
+      completionListOverrideResolver={completionListOverrideResolver}
+      focused={defaultFocused}
+      id={`intellisense-${id}`}
+      scopes={['expressions']}
+      url={getIntellisenseUrl()}
+      value={value || defaultValue}
+      onBlur={props.onBlur}
+      onChange={onChange}
+    >
+      {({ textFieldValue, onValueChanged }) => (
+        <JsonField
+          {...props}
+          style={{ height: 100 }}
+          value={textFieldValue}
+          onBlur={undefined} // onBlur managed by Intellisense
```

Comment:
This is strange, why not remove specifying onBlur?

Reply:
@GeoffCoxMSFT, this is because Intellisense has its own onBlur that looks at both the field and the Intellisense panel. If we don't do undefined on the children then clicking on an Intellisense result would trigger onBlur on the textfield and we would loose focus.

Reply:
Isn't that what event.preventDefault() is for?  If you must pass a value for the property, then pass a no-op function rather than undefined.  You're betting that the React call will pass a value of undefined rather than not passing the property at all.